### PR TITLE
Extend Roadhog Config

### PR DESCRIPTION
**Pull request by whatwewant**
* Features
  * 1 support sass (:heavy_plus_sign: node-sass sass-loader)
  * 2 support webpack resolve alias, add default alias (componets, models, routes, services, utils, assets) for dva (boilerplate), then you can use `import XX from 'components/xxx'` instead of `import XX from '../../components/xxx'` or other.
  * 3 support webpack.ProvidePlugin config, like { "$": "jquery", "React": "react", "_":  "lodash" }, so you donot need to `import` many libraries in you business project.
  * 4 support self output.{filename,chunkFilename}

* How to
  * edit `.roadhogrc`
```
{
   "filename": "[name]-[hash5].js", 
   "chunkFilename": "[name].async.[hash:5].js",
   "alias": {
       "XX": "Path/to/YOUR_DIRECTIRY"
    },
   "provide": {
      "$": "jquery",
      "_": "lodash",
      "React": "react"
    },
    ...
}
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,23 +1,24 @@
 {
-  "name": "roadhog",
-  "version": "1.0.0-beta.7",
-  "description": "Cli tool for serve and build react app, based on create-react-app, support JSON pattern config.",
+  "name": "hyrule",
```
Comment (paranoidjk) [inline package.json]: You shouldn't change the pkg info.

File: src/config/common.js [hunk 2]
```diff
@@ -48,6 +49,16 @@ export function getResolve(config, paths) {
         '.web.js', '.web.jsx', '.web.ts', '.web.tsx',
         '.js', '.json', '.jsx', '.ts', '.tsx',
       ],
+      alias: {
```
Comment (paranoidjk) [inline src/config/common.js]: I am not sure this is a good idea.
- Roadhog is not just for dva project.
- alias do simplify code, but is make vscode go to definition becomes diffcult.

File: package.json (hunk 1 above)
Comment (whatwewant) [inline package.json]: Sorry, this is a accident. I did not expect to affect the PR.

File: src/config/common.js (hunk 2 above)
Comment (whatwewant) [inline src/config/common.js]: Well. the default alias can be removed. In fact, I just want to have a way to define self alias.

**Comment by 104gogo** [comment]
.roadhogrc how to add webpack.ProvidePlugin?

**Comment by sorrycc** [comment]
You can't add webpack plugins in `.roadhogrc`.

**Comment by whatwewant** [comment]
@104gogo Cannot add webpack plugins in `.roadhogrc`, because roadhog want to make it concrolled. Another way, you can do it in single `webpack.config.js` or you can read the source code of roadhog then do something there.  Or see my fork whatwewant/roadhog.

**Comment by 104gogo** [comment]
@whatwewant well, thanks! I had saw roadhog's code, found can both use .roadhogrc and webpack.config.js, so...

**Pull request closed without merging**